ssl/s3_lib.c

```diff
@@ -3770,6 +3769,34 @@ OPENSSL_GLOBAL const SSL_CIPHER ssl3_ciphers[] = {
      256,
      },
 
+    {
+     1,
+     "GOST2012-GOST8912-GOST8912",
+     0x0300ff85,
+     SSL_kGOST,
+     SSL_aGOST12 | SSL_aGOST01,
+     SSL_eGOST2814789CNT12,
+     SSL_GOST89MAC12,
+     SSL_TLSV1,
+     SSL_NOT_EXP | SSL_HIGH,
+     SSL_HANDSHAKE_MAC_GOST12_256 | TLS1_PRF_GOST12_256 | TLS1_STREAM_MAC,
```

Comment:
Dear Matt, 
Unfortunately it does not work this way. 
Now I get an error in the tls1_cert_verify_mac function at line 709.

It is possible to add the GOST extra hashes manually, but it will be the other way to do what I am doing introducing the ssl_get_certverify_digest function. I am not sure it's a good idea.